### DISCUSSION TITLE
Vampire the Masquerade (v20) Character sheet

### DISCUSSION THREAD

**Discussion by silentjames** [Obsidian-TTRPG-Community/fantasy-statblocks#202] [Show and tell]
**Feb 2024 update**:
I moved my development to separate repo, actual files could be found here - https://github.com/silentjames/wod-charsheets-for-obsidian
Now there are not only modern nights' vampires, but also mortal, government agents, werewolves and Dark ages vampires.

---
**Original post**:

Hello folks! First time posting here, and I’d like to share my attempt to recreate Vampire the Masquerade (v20) character sheet with the Fantasy Statblocks plugin. You can see the result on the pics. I’d also like to apologise for any grammatical mistakes I’ll make – English is not my first language. 

First of all, examples in English and Russian:
<img src="https://user-images.githubusercontent.com/12649819/234894483-841cea0d-06ba-4732-a3d7-efeb48124689.png" width="400">   <img src="https://user-images.githubusercontent.com/12649819/234894513-43be6afb-3baf-44f6-bf9f-04ff2cc7bf69.png" width="424.685">

### General things
This statblock is based on custom layout (obviously), css snippet and js-script. To run the script, you’ll need Dataview plugin (but if there is any other way - let me know!). I’ll describe css and js later. And one final warning – I’m not a software developer at all, so css and js are full of sh#tcode :) Again, if something could be improved – let me know! 

### Some restrictions
This was started as small project for me and my friends only, so the final result is not really customisable at some points. For example, it’s made for 680px width, cannot be split into two or more columns. On the other hand, I did my best to hide some rows or blocks if they are empty and/or not important – but you can change it in the layout.
I also based the design on the official Russian version of the character sheet. In general, Russian is 30-40% longer that English, so the statblock could have been not so width… But not yet :) 

### Layout + CSS 
- before the layout, you must put a piece of code for DataviewJS: `dv.executeJs(await dv.io.load("/z_Z manuals/VtMcharlist-en.js"))` with the path to the js-file inside your Obsidian vault (I’m still looking for the solution to run the script from external folder);
- two images (instead of one) can be added, with subtexts;
- “Character sheet” row is collapsible area, closed by default;
- the rule replaced with arrow lines;
- internal links inside some lines, like nature, demeanor, disciplines, flaws and merits, will be displayed like usual text, but still woks;
- all dots and squares are separate fonts. Values from 0 to 9 can be displayed (in the most common cases, 5 or less are enough), but for 10 you must enter X (if you don’t want to use js-script).

### Javascript
What js-script does:
- you can add the name of your character in H2 above the statblock, and it will be hidden in reading view – but remains in the outline panel;
- change the clan image according to the “Clan” property;
- change “Blood Per Turn” value according to “Generation” property;
- all Attributes are equal to 1 by default; 
- if some attributes/abilities/disciplines have more than 5 dots, dots could be replaced to simple numbers, according to the length of the corresponding name; 
- if the path of your Kindred is Humanity, “Normalcy“ will be added as the beading, with corresponding modifier;
- you can now set 10 property value as 10, not X;
- “Blood Pool” area will add new lines of squares if there are more blood than 10, 20, etc, up to 40 as max;
- “Weakness” is also filled automatically, according to the clan.
In general, almost all the points can be done manually (except clan image, I suppose), so the script just automates some things.

### Third-party content
As already been said, it’s based on Russian translation of the game, and because of that I used Marta and TrajanPro3 fonts. Background image is taken from VtM Style Guide from Storytellers Vault; all the dots, squares, damage symbols are also separate fonts; arrows and clan’s logos are vector images. Everything was included in css file as base64 strings – but I’m not sure what can I share because of legal reasons. 
So I removed some “sensitive” content, but there is still a place for them in the files.

### Epilogue 
Well, that’s all, folks! Please let me know what do you think about this layout, and how it can be improved. For example, still not sure about the way the Specialties are displayed, but can’t think up something better.
I also a little bit sad that, in general, this layout is just an info page. Storytelling System assumes that you combine two of your stats, Intelligence + Academics, for example, and then roll all the dice. If you know how it can be achieved – please let me know!

[VtM v20 Vampire en.zip](https://github.com/javalent/fantasy-statblocks/files/11344630/VtM.v20.Vampire.en.zip) - .json and .js
<details>
  <summary>The Layout</summary>

```statblock
columnWidth: 680
layout: VtM v20 Vampire en


name: Edward Lewis 
desc: NSA agent
clan: Nosferatu
generation: 8
sire: [[Duncan Fegredo|Duncan]]
nature: Bon Vivant
demeanor: Competitor
concept: Deviant

img: [[Edward Lewis.png]]
img_title: favorire look
img_alt: [[Edward (real).png]]
img_alt_title: real


Attributes: ATTRIBUTES

physical: PHYSICAL
Strength: 5
Dexterity: 3
Stamina: 4

social: SOCIAL
Charisma: 2
Manipulation: 1
Appearance: 0

mental: MENTAL
Perception: 2
Intelligence: 1
Wits: 2


Abilities: ABILITIES

Talents: TALENTS
alertness: 2
athletics: 1
awareness: 1
brawl: 3
empathy: 0
expression: 5
intimidation: 4
leadership: 4
streetwise: 3
subterfuge: 2

Skills: SKILLS
animalken: 1
crafts: 0
drive: 5
etiquette: 4
firearms: 2
larceny: 4
melee: 3
performance: 2
stealth: 1
survival: 3

Knowledges: KNOWLEDGES
academics: 5
science: 5
law: 4
computer: 4
medicine: 3
occult: 3
politics: 2
investigation: 2
finance: 1
technology: 1


Advantages: ADVANTAGES

Disciplines: DISCIPLINES
discipline1_name: Animalism
discipline1_value: 4
discipline2_name: Obfuscate
discipline2_value: 3
discipline3_name: Potence
discipline3_value: 2
discipline4_name: Celerity
discipline4_value: 1
discipline5_name: Obtenebration
discipline5_value: 1
discipline6_name: Presence
discipline6_value: 1

Backgrounds: BACKGROUNDS
background1_name: Herd
background1_value: 1
background2_name: Fame
background2_value: 2
background3_name: Status
background3_value: 3
background4_name: Resources
background4_value: 4
background5_name: Информаторы
background5_value: 5
background6_name: Domitor
background6_value: 5

Virtues: VIRTUES
conscience: 4
selfcontrol: 2
courage: 3



Merits: MERITS
merit_one_name: Natural Leader
merit_one_value: +1
merit_two_name: Old Pal
merit_two_value: +2
merit_three_name: Friend of the Underground
merit_three_value: +3
merit_four_name: Patagia
merit_four_value: +4
merit_five_name: Arcane
merit_five_value: +5

Flaws: FLAWS
flaw_one_name: Deep Sleeper
flaw_one_value: -1
flaw_two_name: Vulnerability to Silver
flaw_two_value: -2
flaw_three_name: Repelled by Crosses
flaw_three_value: -3
flaw_four_name: Conspicuous Consumption
flaw_four_value: -4
flaw_five_name: Laughingstock
flaw_five_value: -5


path: HUMANITY
path_value: 3

willpower: WILLPOWER
willpower_main: 9
willpower_current: 5

bloodpool: BLOOD POOL
blood: 39


Health: Health
aggravated: 3
lethal: 2
bashing: 2

Specialties: SPECIALTIES
specialty_one_origin: Strength
specialty_one_name: Powerful Arms
specialty_two_origin: Stamina
specialty_two_name: Tough as Nails
specialty_three_origin: Expression
specialty_three_name: Impromptu
specialty_four_origin: Larceny
specialty_four_name: Pickpocketing
specialty_five_origin: 
specialty_five_name: 
specialty_six_origin: 
specialty_six_name: 
```
</details>

**Comment by sigrunixia** [comment]
That is a very sharp (cool, snazzy, crisp!) looking layout. Aesthetically, it is beautiful.

We recently added a javascript block in Statblocks that has access to Obsidian's Global Javascript, that could take over the dataview portion of the JS and make it all internal. It could still access Dataview, if you'd like, but you would not need to have it as an external or custom JS. 

We also added if/else blocks that can sub-block some of the features into smaller _bites_, such as if clan = Goths, Weakness = Emos, Else Sunlight.

**Reply by silentjames** [reply]
Thank so much for your kind words!
Yes, looks like I can add the JavaScript block at the beginning of the layout, so no Dataview is needed! Great.
Need to pay more attention to if/else too - I can probably transfer part of the script to these blocks. 
The biggest "drawback" for me is that I can't combine two properties into single roll - but it looks like feature request for Dice Roller :)

**Comment by lazarusomega2000** [comment]
This looks awesome but I can't seem to get it to work :(. I am not sure what I am doing wrong but I would love some more detail on the css and where the js goes.

**Reply by silentjames** [reply]
Hello! Sorry for the late reply.
So, the steps are simple: 
0. You need Fantasy Statblock installed and enabled, obviously :)
1. You need to import the layout json;
2. You must copy and paste the css file to your "snippets" folder and then enable it in Obsidian's Appearance settings.
3. If you want to use js script, you need to copy it into your vault, and add the following lines into the beginning of the note with statblock:

```
```dataviewjs
dv.executeJs(await dv.io.load("/z_Z manuals/VtMcharlist.js"))
```
In my case, the path to the script is /z_Z manuals/VtMcharlist.js, but you can place it whenever you want.

4. Then copy the layout from the first post into your note, and start filling it!

Hope it will help, let me know if you have any other questions!

**Comment by brad-decker** [comment]
@silentjames this is awesome! Do you care if i use this as a base for a Vampire v5 layout? Also i don't seem to be able to find the .css file is that somewhere I can download?

**Reply by silentjames** [reply]
Thanks mate! Sure you can do it! I was thinking to do the same, with a white marble as background texture; but I don't know v5 good enough (yet).

**Reply by silentjames** [reply]
OMG I cannot believe I could be so stupid =\ Yes, looks like I forgot to add .css to the archive. Well, and this is the reason why @lazarusomega2000 had some troubles. Sorry guys!

**Reply by brad-decker** [reply]
@silentjames no problemo! Do you happen to wanna put this into a repository? I might be able to extract some of your elements in a reusable way for V5 version and then we can collaborate and work on it together. I want to eventually figure out how to modify dice roller and initiative tracker to work for v5/vtm so might be cool to have a place for people to download shared tooling :) In the meantime i'll start working on v5'ing your current layout :D

**Reply by silentjames** [reply]
Hm, I'll definitely give it a try; the problem is that I cannot realize how to work with github properly :D I mean, there is the .css in my snippet folder, and I edit and save the file there to see all changes immediately inside Obsidian. And I don't know how to transfer all the changes to github in the right way, being able to see version history of the file, etc; the only option I see now is just uploading new file every time which is meh.
According to the dice roller: as I understand, now you can only roll one attribute, no options to roll Wits+Occult, for example. And Initiative Tracker is a terra incognita for me :)

**Reply by silentjames** [reply]
Hello @brad-decker!
I'm finally brave enough to create my own repo, in private mode for now. I've sent you the invite, so please come and take a look when you have time! Cheers.

**Comment by silentjames** [comment]
[vtm v20 vampire en v2.zip](https://github.com/javalent/fantasy-statblocks/files/13267115/vtm.v20.vampire.en.v2.zip)

Hi all, here is the new version of the character sheet! Was continue working on it from time to time since the first post; can't remember what exactly I changed\added... At least, it now supports Dice Roller :D
Here is the layout:
<details>
  <summary>The Layout</summary>

```
```dataviewjs 
dv.executeJs(await dv.io.load("/z_Z manuals/VtMcharlist-en.js"))
```
```

```statblock
columnWidth: 680
layout: VtM v20 Vampire en

name: Edward Lewis 
desc: NSA agent
clan: Tzimisce
generation: 8
sire: [[Duncan Fegredo|Duncan]]
nature: Bon Vivant
demeanor: Competitor
concept: Deviant

img: [[Валентайн.png]]
img_title: favorire look
img_alt: [[Руфус (настоящий).png]]
img_alt_title: real


Attributes: ATTRIBUTES

Physical: PHYSICAL
Strength: 5
Dexterity: 3
Stamina: 4

Social: SOCIAL
Charisma: 2
Manipulation: 1
Appearance: 0

Mental: MENTAL
Perception: 2
Intelligence: 1
Wits: 2


Abilities: ABILITIES

Talents: TALENTS
alertness: 2
athletics: 1
awareness: 1
brawl: 3
empathy: 0
expression: 5
intimidation: 4
leadership: 4
streetwise: 3
subterfuge: 2

Skills: SKILLS
animalken: 1
crafts: 0
drive: 5
etiquette: 4
firearms: 2
larceny: 4
melee: 3
performance: 2
stealth: 1
survival: 3

Knowledges: KNOWLEDGES
academics: 5
science: 5
law: 4
computer: 4
medicine: 3
occult: 3
politics: 2
investigation: 2
finance: 1
technology: 1


Advantages: ADVANTAGES

Disciplines: DISCIPLINES
discipline1_name: Animalism
discipline1_value: 4
discipline2_name: Obfuscate
discipline2_value: 3
discipline3_name: Potence
discipline3_value: 2
discipline4_name: Celerity
discipline4_value: 1
discipline5_name: Obtenebration
discipline5_value: 1
discipline6_name: Presence
discipline6_value: 1

Backgrounds: BACKGROUNDS
background1_name: Herd
background1_value: 1
background2_name: Fame
background2_value: 2
background3_name: Status
background3_value: 3
background4_name: Resources
background4_value: 4
background5_name: Информаторы
background5_value: 5
background6_name: Domitor
background6_value: 5

Virtues: VIRTUES
conscience: 4
selfcontrol: 2
courage: 3



Merits: MERITS
merit1_name: Natural Leader
merit1_value: -1
merit2_name: Old Pal
merit2_value: -2
merit3_name: Friend of the Underground
merit3_value: -3
merit4_name: Patagia
merit4_value: -4
merit5_name: Arcane
merit5_value: -5

Flaws: FLAWS
flaw1_name: Deep Sleeper
flaw1_value: +1
flaw2_name: Vulnerability to Silver
flaw2_value: +2
flaw3_name: Repelled by Crosses
flaw3_value: +3
flaw4_name: Conspicuous Consumption
flaw4_value: +4
flaw5_name: Laughingstock
flaw5_value: +5


path: HUMANITY
path_value: 3

willpower: WILLPOWER
willpower_main: 9
willpower_current: 5

bloodpool: BLOOD POOL
blood: 39


Health: HEALTH
aggravated: 3
lethal: 2
bashing: 2

Specialties: SPECIALTIES
specialty1_origin: Strength
specialty1_name: Powerful Arms
specialty2_origin: Stamina
specialty2_name: Tough as Nails
specialty3_origin: Expression
specialty3_name: Impromptu
specialty4_origin: Larceny
specialty4_name: Pickpocketing
specialty5_origin: 
specialty5_name: 
specialty6_origin: 
specialty6_name: 

show_weakness: yes
```
</details>

**Reply by lazarusomega2000** [reply]
Allrighty, I get some weird results from this thing. I almost got it to work and then it looked weird again... Here is my screen cap of the results of loading this thing in... I don't know why it can't find your CSS. I have it activated....
![image](https://github.com/javalent/fantasy-statblocks/assets/32077293/5a08140e-dec4-452c-b044-311f97a55b73)

**Reply by lazarusomega2000** [reply]
Alrighty, dashes and spaces mean something but I am still not sure what's up with the rest. I have it this far. Am I going to have to figure out all the pictures and fonts to load for it to find?
![image](https://github.com/javalent/fantasy-statblocks/assets/32077293/24823f1a-25d5-408d-aa59-632e03ba52be)

**Reply by lazarusomega2000** [reply]
Okay, I figured out that my theme was messing with it, and adding columns: 1 to the layout solved a few issues for me in particular as it kept splitting it into two columns no matter what else I did. I widened it out to 780 and it seems to fit for that, but there's still no clan background and the bloodpoints/etc. don't do what they're supposed to...
![image](https://github.com/javalent/fantasy-statblocks/assets/32077293/27b1cc4e-d0d1-4b36-a800-c5c59cc9b1ba)

**Reply by silentjames** [reply]
Hey @lazarusomega2000! Sorry for the late reply mate.

Which theme do you use? ITS Theme? I'm mostly using Things, but will also take a look at this one.

Yes, this layout is only adapted for `column: 1` and `columnWidth: 680`. Am I right that you have tried 680 and that also looks weird?

The js script is required for clan backgrounds (couldn't find another solution, unfortunately), and it is also needed in some cases to display humanity\willpower\blood values correctly. 
Let me explain this: I created the hand-made fonts to replace numbers with circles and squares there - it works perfectly for single digits, but not for "10". So, if you don't want to use the script, you should type "X" instead of "10" for humanity and willpower; for blood, if you character is 12th generation or higher and his bloodpool is, for example, 15, you should use two fields: `blood: X` and `blood_current2: 5`. There is also `blood_current3` and `blood_current4`, in case you need a powerful elder.
If you use the script, it will automatically do all described above and a little bit more - clan backgrounds, set up blood per turn value based on the generation, set up bearing value bases on the humanity (if the path is humanity ofc), and some other small things. 

About the fonts: I'm not sure, but if you install Marta and TrajanPro3 fonts on your computer, it will probably be enough, Obsidian will use them from the system.
If not, there are some places for them in the css, like this:
```
@font-face {
    font-family: 'Marta';
    font-weight: normal;
    src: /*  base64 string as url  */ format('truetype');
}
```
You need to get the font file from anywhere, then encode the ttf file to base64 string. I use [this site](https://base64.guru/converter/encode/file) - select your font file, choose "Data URI" as output format and press the big button. You'll get a looong wall of random numbers. That's what you need.
Go back to css, remove `/*  base64 string as url  */`, type instead `url("")`, and paste this long encoded text between quotes. So you'll get something like:
```src: url(data:font/truetype;charset=utf-8;base64,AAEAAAANAIAAAwBAAAAHEdERUYAKQBrAACDrAA format('truetype');``` 
Way longer, of course. Save changes, and you're done!
You need to repeat this for all the fonts.

Hope I didn't forget anything and described all steps correctly. If you do have any other questions - feel free to ask!

**Reply by lazarusomega2000** [reply]
So I have the jsscript in there, but it's not working. I am not sure why it's not, but it's not.

**Reply by silentjames** [reply]
Hm, that's strange. I could catch the bug from time to time with a clan background not being displayed, but other small functions should be fine.
How do you use it? Here is the example from my note:
<img width="656" alt="image" src="https://github.com/javalent/fantasy-statblocks/assets/12649819/e3482212-c833-4b18-a4b0-f21734561087">
Is there any errors - in the note or in the console? You can find the console by pressing Alt + Command + i, then go to console tab.

**Comment by silentjames** [comment]
I moved my development to separate repo, actual files could be found here - https://github.com/silentjames/wod-charsheets-for-obsidian
Now there are not only modern nights' vampires, but also mortal, government agents, werewolves and Dark ages vampires.

**Comment by Lancaban** [comment]
I am having issues with the file browser becoming very sluggish/laggy after opening one of the sheets.

If I close and restart Obsidian, it will preform just fine, until such time as I try to open a note containing the sheet. Once I do this, dragging and dropping or even mousing over of files in the file-browser becomes almost impossible. I can move the mouse, and the tooltip and actual file will lag almost a minute behind actual movement.
